internal/gitserver/connection/addrs.go

```diff
@@ -218,4 +222,9 @@ func (a *atomicGitServerConns) update(cfg *conf.Unified) {
 	}
 }
 
+var (
+	concurrencyLimit = env.MustGetInt("SRC_GITSERVER_CLIENT_CONCURRENCY_LIMIT", 500, "maximum number of concurrent gitserver client connections")
```

Comment:
This only applies to RPCs, not other kinds of calls like git clones. You should tweak the message to reflect this.

```suggestion
	concurrencyLimit = env.MustGetInt("SRC_GITSERVER_CLIENT_CONCURRENCY_LIMIT", 500, "maximum number of concurrent gitserver RPC calls")
```

Reply:
I was referring to gitserver.Client here - which only does RPC calls, but I'm with with this update if it's clearer